src/natcap/invest/coastal_blue_carbon/preprocessor.py

```diff
@@ -76,292 +67,175 @@ def execute(args):
+
+    try:
+        n_workers = int(args['n_workers'])
+    except (KeyError, ValueError, TypeError):
+        # KeyError when n_workers is not present in args
+        # ValueError when n_workers is an empty string.
+        # TypeError when n_workers is None.
+        n_workers = -1  # Synchronous mode.
+    task_graph = taskgraph.TaskGraph(
+        taskgraph_cache_dir, n_workers, reporting_interval=5.0)
+
+    snapshots_dict = (
+        coastal_blue_carbon._extract_snapshots_from_table(
+            args['landcover_snapshot_csv']))
+
+    # Align the raster stack for analyzing the various transitions.
+    min_pixel_size = float('inf')
+    source_snapshot_paths = []
+    aligned_snapshot_paths = []
+    for snapshot_year, raster_path in snapshots_dict.items():
+        source_snapshot_paths.append(raster_path)
+        aligned_snapshot_paths.append(os.path.join(
+            output_dir, ALIGNED_LULC_RASTER_TEMPLATE.format(
+                year=snapshot_year, suffix=suffix)))
+        min_pixel_size = min(
+            utils.mean_pixel_size_and_area(
+                pygeoprocessing.get_raster_info(raster_path)['pixel_size'])[0],
+            min_pixel_size)
+
+    baseline_srs_wkt = pygeoprocessing.get_raster_info(
+        snapshots_dict[min(snapshots_dict.keys())])['projection_wkt']
+    alignment_task = task_graph.add_task(
+        func=pygeoprocessing.align_and_resize_raster_stack,
+        args=(source_snapshot_paths,
+              aligned_snapshot_paths,
+              (['near']*len(source_snapshot_paths)),
+              (min_pixel_size, -min_pixel_size),
+              'intersection'),
+        kwargs={'target_projection_wkt': baseline_srs_wkt},
+        hash_algorithm='md5',
+        copy_duplicate_artifact=True,
+        target_path_list=aligned_snapshot_paths,
+        task_name='Align input landcover rasters')
+
+    landcover_table = utils.build_lookup_from_csv(
+        args['lulc_lookup_table_path'], 'code')
+
+    target_transition_table = os.path.join(
+        output_dir, TRANSITION_TABLE.format(suffix=suffix))
+    _ = task_graph.add_task(
+        func=_create_transition_table,
+        args=(landcover_table,
+              sorted(snapshots_dict.values(), key=lambda x: [0]),
+              target_transition_table),
+        target_path_list=[target_transition_table],
+        dependent_task_list=[alignment_task],
+        task_name='Determine transitions and write transition table')
+
+    # Creating this table should be cheap, so it's not in a task.
+    # This is only likely to be expensive if the user provided a landcover
+    # lookup with many many rows. If that happens, this tool will have other
+    # problems (like exploding memory in the transition table creation).
+    target_biophysical_table_path = os.path.join(
+        output_dir, BIOPHYSICAL_TABLE.format(suffix=suffix))
+    _create_biophysical_table(landcover_table, target_biophysical_table_path)
```

Comment:
Just to note (I don't think this is the case here) but an advantage of creating a file in TaskGraph even if it's fast is if that file were used by a task later. If you recreate it every time then the task down the line will also reexecute every time.

Reply:
Sounds good to me!  In this case, there aren't any downstream tasks, but I don't have an otherwise good reason to keep it out of a task.  Wrapped in e5749c40